### PR TITLE
Fishing map/fetch context geometry

### PR DESCRIPTION
**Pull request by j8seangel**
Since the tiles clip the feature geometries the geometry to perform the analysis is now fetched from the 
`/v1/datasets/:dataset/user-context-layer-v1/:gfw_id` endpoint.

This PR does 2 things:
1. save the datasetId into the feature + the url
2. moves the logic from `useAnalysisGeometry` to `fetchAnalysisAreaThunk` allowing to keep the rest almost the same

TODO:
- [x] Download geometry for reports too
- [x] Fix analysis name description from drawn or custom geometries

### CODE REVIEW

File: apps/fishing-map/features/map/popups/ContextLayers.hooks.ts [hunk 1]
```diff
@@ -48,12 +46,13 @@ export const useContextInteractions = () => {
         return
       }
       batch(() => {
-        dispatch(
-          setDownloadActivityGeometry({
-            geometry: feature.geometry,
-            name: feature.value || feature.title,
-          })
-        )
+        // TODO get geometry from API too
```
Comment (nerik) [inline apps/fishing-map/features/map/popups/ContextLayers.hooks.ts:49]: This needs to be done before merging right?

File: libs/api-types/src/geometries.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import type { MultiPolygon, Polygon } from 'geojson'
+
+export interface ContextAreaGeometryProperties {
+  fid: string
+  mrgid: string
```
Comment (nerik) [inline libs/api-types/src/geometries.ts:5]: Are `fid` and `mrgid` actually needed at some point? What do they represent?

**Review by nerik — COMMENTED** [review]
Screenshot 2022-01-21 at 10 05 41
Looking good! 😊

File: libs/api-types/src/geometries.ts (hunk 2 above)
Comment (j8seangel) [reply libs/api-types/src/geometries.ts:5]: Nop, they are not used anywhere but as they are in the api response I wanted to type properly

File: apps/fishing-map/features/map/popups/ContextLayers.hooks.ts (hunk 1 above)
Comment (j8seangel) [reply apps/fishing-map/features/map/popups/ContextLayers.hooks.ts:49]: Yes, working on it now

File: apps/fishing-map/features/analysis/analysis.hooks.ts [hunk 3]
```diff
@@ -159,38 +147,36 @@ export const useAnalysisGeometry = () => {
 
   const setAnalysisBounds = useCallback(
     (bounds: Bbox) => {
-      dispatchQueryParams({ analysis: { areaId, bounds, sourceId } })
+      dispatchQueryParams({ analysis: { areaId, bounds, sourceId, datasetId } })
```
[the remark this answers is not in the document]
Comment (j8seangel) [reply apps/fishing-map/features/analysis/analysis.hooks.ts:150]: Nop, but will have a workaround. Good catch!

File: apps/fishing-map/features/areas/areas.slice.ts [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+      }
+    })
+    builder.addCase(fetchAreaThunk.rejected, (state, action) => {
+      const key = getAreaKey(action.meta.arg)
+      state[key] = { ...state[key], status: AsyncReducerStatus.Error }
+    })
+  },
+})
+
+export const { clearAreaGeometry } = areasSlice.actions
```
[the remark this answers is not in the document]
Comment (j8seangel) [reply apps/fishing-map/features/areas/areas.slice.ts:80]: Nop, I've just removed and thanks!

File: apps/fishing-map/features/analysis/analysis.hooks.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (j8seangel) [reply apps/fishing-map/features/analysis/analysis.hooks.ts:155]: Fixed at: https://github.com/GlobalFishingWatch/frontend/commit/be2d3ac8f02d62ebfc5df4d5635ca5305098578a

**Pull request merged**